File: ios/RNWireguard.m
#if __has_include("RCTBridgeModule.h")
#import "RCTBridgeModule.h"
#else
#import <React/RCTBridgeModule.h>
#endif

@interface RCT_EXTERN_MODULE(RNWireguard, NSObject)

RCT_EXTERN_METHOD(Connect: (NSString *)config session: (NSString *)session notif: (NSString *)notif resolve: (RCTPromiseResolveBlock)resolve reject: (RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(Disconnect: (RCTPromiseResolveBlock)resolve reject: (RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(Version: (RCTPromiseResolveBlock)resolve reject: (RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(Status: (RCTPromiseResolveBlock)resolve reject: (RCTPromiseRejectBlock)reject)

@end

File: ios/RNWireguard.swift
import Foundation

@objc(RNWireguard)
class RNWireguard: NSObject {
    @objc
    func constantsToExport() -> [String: Any]! {
        return ["EV_TYPE_EXCEPTION": "EV_TYPE_EXCEPTION",
            "EV_TYPE_REGULAR": "EV_TYPE_REGULAR",
            "EV_REVOKED": "EV_REVOKED",
            "EV_HOST_DESTROYED": "EV_HOST_DESTROYED",
            "EV_HOST_RESUMED": "EV_HOST_RESUMED",
            "EV_HOST_PAUSED": "EV_HOST_PAUSED"]
    }

    @objc
    static func requiresMainQueueSetup() -> Bool {
        return false
    }

    @objc(Connect:session:notif:resolve:reject:)
    func Connect(config: String, session: String, notif: String, resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
        reject("x", "x", nil)
    }

    @objc(Disconnect:reject:)
    func Disconnect(resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
        reject("x", "x", nil)
    }

    @objc(Version:reject:)
    func Version(resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
        reject("x", "x", nil)
    }

    @objc(Status:reject:)
    func Status(resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
        reject("x", "x", nil)
    }
}
